mcodingbot/utils/peps.py

```diff
@@ -39,10 +41,19 @@ async def fetch_pep_info(self, bot: Bot) -> None:
                     )
                     for pep_id, pep in (await resp.json()).items()
                 }
+                self._pep_map = {
+                    k: f"{v.title} ({v.number})" for k, v in self._peps.items()
+                }
 
     def get(self, pep_number: int) -> PEPInfo | None:
         return self._peps.get(pep_number)
 
+    def search(
+        self, query: str, limit: int = 20
+    ) -> Generator[PEPInfo, None, None]:
+        res = find(query, self._pep_map, limit=limit)
+        return (self.get(pep[2]) for pep in res)
```

Comment:
Don't forget to fix this part (this is why mypy is failing). One way to fix it is like this:

```py
return (
    pep_info
    for pep in res
    if (pep_info := self.get(pep[2])) is not None
)